tests/sla_tests.js

```diff
@@ -429,4 +465,180 @@ describe('SLA Bridge and Channels Tester', function() {
       }, asyncDelay);
     } 
   });
+  it('should mark a device as RINGING when dialing an outbound channel',
+      function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    config='tests/testConfigs/singleEndpoint.json';
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+    client.deviceStates.update({deviceName: 'Stasis:42',
+      deviceState: 'NOT_INUSE'}, function() {});
+
+    markAsRinging();
+    function markAsRinging() {
+      setTimeout(function() {
+        if (isMixing && bridges.length === 1 && inbound.inbound &&
+          dialed[0] && mockDeviceStates[0].hasRung &&
+          mockDeviceStates[0].deviceName === 'Stasis:42') {
+            done();
+          } else {
+            markAsRinging();
+          }
+      }, asyncDelay);
+    }
+  });
+  it('should mark a device as RINGING when dialing multiple outbound channels',
+      function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    config='tests/testConfigs/multipleEndpoints.json';
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+    client.deviceStates.update({deviceName: 'Stasis:42',
+      deviceState: 'NOT_INUSE'}, function() {});
+
+    markAsRinging();
+    function markAsRinging() {
+      setTimeout(function() {
+        if (isMixing && bridges.length === 1 && inbound.inbound &&
+          dialed[0] && dialed[1] && mockDeviceStates[0].hasRung &&
+          mockDeviceStates[0].deviceName === 'Stasis:42') {
+            done();
+        } else {
+          markAsRinging();
+        }
+      }, asyncDelay);
+    }
+  });
+  it('should give the application an invalid configuration file and promptly ' +
+      'fail out', function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    config = 'tests/testConfigs/invalid.json';
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+
+    invalidConfigurationFile();
+    function invalidConfigurationFile() {
+      setTimeout(function() {
+        if (configurationFailed) {
```

Comment:
Typically, using setTimeout to check local variables for state during a test is done because you have no direct control over the state you want to check against.

It would be better here to simply provide your own anonymous function to the catch handler instead of relying on errHandler. You could then call done() inside your anonymous function if the error thrown is the one you expect.